odl/space/cu_ntuples.py

```diff
@@ -943,8 +1318,14 @@ def norm(self, x):
         norm : float
             The norm of the vector
         """
-        from math import sqrt
-        return sqrt(self.const) * float(_norm_default(x))
+        if self.exponent == 2.0:
+            from math import sqrt
+            return sqrt(self.const) * float(_norm_default(x))
+        elif self.exponent == float('inf'):  # Weighting irrelevant
```

Comment:
That one could do in principle it would define a valid norm, but the question is what we prioritize - mathematical consistency in the sense I sketched above, with the potentially surprising implication that the inf-norm does not care about weighting - or to save users from surprises. Not really sure.